res/controllers/Numark-Scratch-scripts.js

```diff
@@ -0,0 +1,430 @@
+            input: inputUnshifted,
+            shift: function() {
+                this.input = inputShifted;
+            },
+            unshift: function() {
+                this.input = inputUnshifted;
+            },
+        });
+    }
+
+    this.paddle = new components.Button({
+        group: "[EffectRack1_EffectUnit" + deckNumber + "]",
+        outKey: "enabled",
+        inKey: "group_[Channel" + deckNumber + "]_enable",
+        input: function(channel, control, value, _status, _group) {
+            this.isPress = value !== 0;
+            engine.setValue("[EffectRack1_EffectUnit1]", "group_[Channel" + deckNumber + "]_enable", this.isPress);
+            engine.setValue("[EffectRack1_EffectUnit2]", "group_[Channel" + deckNumber + "]_enable", this.isPress);
+            this.trigger();
+        }
+    });
+
+    this.dryWetKnob = new components.Pot({
+        group: "[EffectRack1_EffectUnit" + deckNumber + "]",
+        inKey: "mix",
+        shift: function() {
+            this.inKey = "super1";
+        },
+        unshift: function() {
+            this.inKey = "mix";
+        },
+    });
+};
+NumarkScratch.EffectUnit.prototype = new components.ComponentContainer();
+
+NumarkScratch.XfaderContainer = function() {
+    this.crossfader = new components.Pot({
+        midi: [0xBF, 0x08],
+        group: "[Master]",
+        inKey: "crossfader",
+    });
+
+    this.setCurve = function(channel, control, value, _status, _group) {
+        switch (value) {
+        case 0x00: // Additive/Linear
+            engine.setValue("[Mixer Profile]", "xFaderMode", 0);
+            engine.setValue("[Mixer Profile]", "xFaderCalibration", 0.4);
+            engine.setValue("[Mixer Profile]", "xFaderCurve", 0.9);
+            break;
+        case 0x7F:  // Picnic Bench/Fast Cut
+            engine.setValue("[Mixer Profile]", "xFaderMode", 0);
+            engine.setValue("[Mixer Profile]", "xFaderCalibration", 0.9);
+            engine.setValue("[Mixer Profile]", "xFaderCurve", 7.0);
+            break;
+        }
+    };
+
+    this.xFaderReverse = function(channel, control, value, _status, _group) {
+        // 0x7F is ON, 0x00 is OFF
+        engine.setValue("[Mixer Profile]", "xFaderReverse", (value === 0x7F) ? 1 : 0);
+    };
+};
+
+NumarkScratch.XfaderContainer.prototype = new components.ComponentContainer();
+
+NumarkScratch.setChannelInput = function(channel, control, value, _status, _group) {
+    const number = (control === 0x57) ? 1 : 2;
+    const channelgroup = "[Channel" + number + "]";
+
+    switch (value) {
+    case 0x00:  // PC and turn on vinyl control
+        engine.setValue(channelgroup, "passthrough", 0);
+        engine.setValue(channelgroup, "vinylcontrol_enabled", 1);
+        break;
+    case 0x02:  // PHONO/LINE and turn off vinyl control
+        engine.setValue(channelgroup, "passthrough", 1);
+        engine.setValue(channelgroup, "vinylcontrol_enabled", 0);
+        break;
+    }
+};
```

Comment:
@JoergAtGithub I noticed you have similar code on your Reloop KUT mapping. For some reason on Mixxx startup, Mixxx is not reading the status of the switch even though I get a sysex response. I have to physically toggle the switch, after Mixxx has loaded for the status it to be read.

Could it be my code or is Mixxxx loading too quickly? Would appreciate your help. Besides this, the PR is ready for merging.

Reply:
> Could it be my code or is Mixxx loading too quickly? 

If you start mix first with the mapping disabled and then enable it later via the preferences, does it work reliably? 
fyi I also have encountered a firmware bug on my NS6II where it sometimes sends the wrong switch value (though this was with the split cue toggle, not the input switches).

Reply:
if it works with `--qml`, it might indeed be some sort of data race on startup. In that case, I'd call it acceptable to slightly delay the sysex inquiry message.

Reply:
@Swiftb0y a 9 second delay works. Seems a little too long imo. Is that acceptable?

the 9sec correlates to how long it takes for Mixxx to open up for me, on Windows.

Reply:
> > Could it be my code or is Mixxx loading too quickly?
> 
> If you start mix first with the mapping disabled and then enable it later via the preferences, does it work reliably? fyi I also have encountered a firmware bug on my NS6II where it sometimes sends the wrong switch value (though this was with the split cue toggle, not the input switches).

@Swiftb0y , if I start Mixxx first with the mapping disabled and then enable it later via the preferences, it work reliably.
If I also reload the mapping via text editor after Mixxx has fully loaded, it works reliably.